### ISSUE TITLE
Optionally delete the avro schema in schema registry when a stream/table is dropped.

### ISSUE THREAD

**Issue by hjafarpour** [confluentinc/ksql#1189]
If we have an avro stream or table, we used to not clean up the schema registry after dropping the stream/table. After the schema registry PR merge, we now delete the avro schema for a stream/table after we drop the stream/table. This may not be desirable behavior especially if the topic for the stream/table existed before or there are external users of this topic other than KSQL.
One way to avoid accidentally deleting the schema of a topic is to have user explicitly indicate that the schema for the topic should be deleted too. 
We can add an optional suffix, `WITH AVRO SCHEMA`, to `DROP STREAM/TABLE` statement so user can explicitly specify the desired behavior. 
If user just wants to drop the stream/table without deleting the corresponding schema from schema registry `DROP` statement will be used without any change:

`ksql> DROP STREAM foo;`

If user wants to delete the associated schema in the schema registry when dropping a stream/table the following statement will be used:

`ksql> DROP STREAM foo WITH AVRO SCHEMA;`

The above statement will drop the stream and also delete the corresponding schema from the schema registry.

This change will only affect streams/tables with AVRO format that use schema registry and it won't affect the JSON and DELIMITED formats.

**Comment by guozhangwang** [comment]
Personally I think the dropping of the schema can be aligned with the deletion of the sink topics: we will only delete the schema when users specify to delete the topics as well; and we do not allow deleting schemas when the topic is still there. 

If we allow users to delete schema while dropping the stream / table, but not deleting the topics such that topics may be reused later. Then there is a risk that users can register a new schema which bypass any compatibility checks on the SR server. What are the scenarios where we do want to open this door to the users?

**Comment by hjafarpour** [comment]
Yes, I agree. Indeed I just had  a discussion with @rodesai on letting user to delete topic when dropping a stream/table. That way we can delete the mete data associated with the topic too. 
We can have similar syntax such as `DROP STRAM foo WITH TOPIC;` to indicate that the topic should also be deleted. I'll create another issue for deleting topic and refer to this issue.
